File: dune/xt/grid/walker.cc
// This file is part of the dune-xt-grid project:
//   https://github.com/dune-community/dune-xt-grid
// The copyright lies with the authors of this file (see below).
// License: Dual licensed as BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
//      or  GPL-2.0+ (http://opensource.org/licenses/gpl-license)
//          with "runtime exception" (http://www.dune-project.org/license.html)
// Authors:
//   Felix Schindler (2017)

#include "config.h"

#include "walker.hh"

#if HAVE_DUNE_PYBINDXI

namespace Dune {
namespace XT {
namespace Grid {


template class Walker<
    typename Layer<YaspGrid<2, EquidistantOffsetCoordinates<double, 2>>, Layers::leaf, Backends::view>::type>;
template class Walker<
    typename Layer<YaspGrid<2, EquidistantOffsetCoordinates<double, 2>>, Layers::level, Backends::view>::type>;
#if HAVE_DUNE_FEM
template class Walker<
    typename Layer<YaspGrid<2, EquidistantOffsetCoordinates<double, 2>>, Layers::leaf, Backends::part>::type>;
template class Walker<
    typename Layer<YaspGrid<2, EquidistantOffsetCoordinates<double, 2>>, Layers::level, Backends::part>::type>;
#endif // HAVE_DUNE_FEM


#if HAVE_ALUGRID
template class Walker<typename Layer<ALUGrid<2, 2, simplex, conforming>, Layers::leaf, Backends::view>::type>;
template class Walker<typename Layer<ALUGrid<2, 2, simplex, conforming>, Layers::level, Backends::view>::type>;
#if HAVE_DUNE_FEM
template class Walker<typename Layer<ALUGrid<2, 2, simplex, conforming>, Layers::leaf, Backends::part>::type>;
template class Walker<typename Layer<ALUGrid<2, 2, simplex, conforming>, Layers::level, Backends::part>::type>;
#endif // HAVE_DUNE_FEM
#endif // HAVE_ALUGRID


} // namespace Grid
} // namespace XT
} // namespace Dune

#endif // HAVE_DUNE_PYBINDXI

File: python/test/walker.py
# ~~~
# This file is part of the dune-xt project:
#   https://github.com/dune-community/dune-xt
# Copyright 2009-2020 dune-xt developers and contributors. All rights reserved.
# License: Dual licensed as BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
#      or  GPL-2.0+ (http://opensource.org/licenses/gpl-license)
#          with "runtime exception" (http://www.dune-project.org/license.html)
# Authors:
#   René Fritze    (2018 - 2019)
#   Tobias Leibner (2018 - 2020)
# ~~~

from pymortests.base import runmodule

from dune.xt.grid import Dim, Cube, Simplex, make_cube_grid, Walker

init_args = (
    (Dim(1), [0], [1], [2]),
    (Dim(2), Cube(), [0, 0], [1, 1], [2, 2]),
    (Dim(2), Simplex(), [0, 0], [1, 1], [2, 2]),
    (Dim(3), Cube(), [0, 0, 0], [1, 1, 1], [2, 2, 2]),
    (Dim(3), Simplex(), [0, 0, 0], [1, 1, 1], [2, 2, 2]),
)


def test_init():
    for args in init_args:
        grid = make_cube_grid(*args)
        walker = Walker(grid)

def test_walk():
    for args in init_args:
        grid = make_cube_grid(*args)
        walker = Walker(grid)
        walker.walk()

if __name__ == "__main__":
    runmodule(filename=__file__)

